sled-agent/config-reconciler/src/reconciler_task/zones.rs

```diff
@@ -706,6 +710,16 @@ impl OmicronZone {
             ));
         }
 
+        // Make a best effort to archive the zone's log files.
+        if let Some(zone_dataset_root) = running_zone.root().parent() {
```

Comment:
I'm vaguely surprised we need `.parent()` here - is that because `root()` returns the path we need plus `/root` on the end?